### PR TITLE
fix: reference the correct item in item selected event

### LINKED ISSUE #220: MenuBar and ContextMenu onItemSelected uses invalid event type
The `event.detail.value` of `onItemSelected` event in `MenuBar` and `ContextMenu` isn't compatible with the updated `MenuBarItem` type.

Regression from https://github.com/vaadin/react-components/pull/205

Screenshot 2024-01-15 at 15 46 17

### PR DESCRIPTION
**Pull request by tomivirkki**
## Description

Fixes #220 

## Type of change

Bugfix

### CODE REVIEW

File: src/MenuBar.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
   type MenuBarItem as _MenuBarItem,
   type SubMenuItem as _SubMenuItem,
 } from './generated/MenuBar.js';
-import { mapItemsWithComponents } from './utils/mapItemsWithComponents.js';
+import { getOriginalItem, mapItemsWithComponents } from './utils/mapItemsWithComponents.js';
 
 export * from './generated/MenuBar.js';
```
Comment (tomivirkki) [inline src/MenuBar.tsx:11]: Looks like the exported `MenuBarItemSelectedEvent` also needs to be updated. Working on it.

File: src/utils/mapItemsWithComponents.ts [hunk 2]
```diff
@@ -40,16 +41,22 @@ export function mapItemsWithComponents<T>(
         ...rest,
         component: root,
         children: webComponentChildren,
+        __item: item,
```
Comment (web-padawan) [inline src/utils/mapItemsWithComponents.ts:44]: This line doesn't seem to be covered by tests added in the PR, as they only cover `else` block.

File: src/MenuBar.tsx (hunk 1 above)
Comment (tomivirkki) [reply src/MenuBar.tsx:11]: Done

File: src/utils/mapItemsWithComponents.ts (hunk 2 above)
Comment (tomivirkki) [reply src/utils/mapItemsWithComponents.ts:44]: Updated the test case

**Review by web-padawan — APPROVED** [review]

**Pull request merged**